### ISSUE TITLE
Optimonal Upload to a Vector Database

### ISSUE THREAD

**Issue by abimek** [bbawj/obsidian-semantic-search#4]
Hello! 

I've been meaning to create an obsidian plugin to upload my notes/embeddings to a vector database, I'm wondering if this is a feature that you would want in your plugin, if so let me know and I'll create it and make a PR. I was thinking about supporting Pinecone initally and adding support for other systems later on.

**Comment by bbawj** [comment]
Hi! Just so I understand you correctly, are you suggesting adding a feature to allow pulling embedding data from Pinecone rather than from an embedding.csv file as it is currently done?

**Comment by abimek** [comment]
More like upload your embeddings to a pinecone database, although now that you mention that, I think that would be a fantastic feature as well. Maybe just full pinecone / vector db integration.

**Comment by abimek** [comment]
and by full integration i mean both upload and download

**Comment by bbawj** [comment]
Yes that sounds quite useful. Please open a PR if you are able to support this!

**Comment by abimek** [comment]
Development on this will take longer then I expected, I extracted the underlying rust implementation of the pinecone api and had everything complete but once I did wasm wouldn't build it because it was using tcp and tokio. I'll be making my own implementation and then will come back to this in a couple days.

**Comment by abimek** [comment]
Well this was a dead end, I made a crate to support pinecone requests, both wasm and non wasm https://github.com/abimek/pinenut but guess what. Pinecone handles CORS weirdly so anything coming from localhost won't work.

**Comment by bbawj** [comment]
Ah but does this mean that it can support instances not on localhost?

**Comment by abimek** [comment]
Yes, I believe it reads localhost as different IP addresses but a static ip address will work. You can read up on it more here https://support.pinecone.io/hc/en-us/articles/8764985175709-CORS-Issues I was thinking of making some kid of proxy but that won't work since the proxy might get overloaded over time. It would be a fun project to make a distributed one that scales based on demand though. Hmmm. I probably can't afford that though.
